### ISSUE TITLE
Arch Linux: systemctl suspend + physlock failing

### ISSUE THREAD

**Issue by fgro** [muennich/physlock#28]
Hi folks,

I've been trying to get Arch Linux systemd suspend to work with physlock, unfortunately with no success. I tried to setup my service files according to https://github.com/muennich/physlock/issues/10 

My physlock version

```
% physlock -v
physlock 0.4.5 - lock all consoles
```
Here is my service template enabled as "suspend@fgro.service":

```
[Unit]
Description=Lock X session
Before=sleep.target

[Service]
User=%I
Type=simple
Environment=DISPLAY=:0
ExecStart=/usr/bin/physlock -ds
ExecStartPost=/usr/bin/sleep 2

[Install]
WantedBy=sleep.target
```

Here is the log of a "systemctl suspend" and "resume" process:

https://gist.github.com/anonymous/de00e680db4afcb57871

I think the interesting lines are 88-103:

```
Nov 06 12:00:08 triton systemd[1]: suspend@fgro.service: Main process exited, code=exited, status=1/FAILURE
Nov 06 12:00:08 triton avahi-daemon[531]: Leaving mDNS multicast group on interface wwp0s20u3i8.IPv6 with address fe80::7dbd:2a03:fc3a:ba45.
Nov 06 12:00:08 triton systemd[1]: suspend@fgro.service: Unit entered failed state.
Nov 06 12:00:08 triton avahi-daemon[531]: Withdrawing address record for fe80::7dbd:2a03:fc3a:ba45 on wwp0s20u3i8.
Nov 06 12:00:08 triton systemd[1]: suspend@fgro.service: Failed with result 'exit-code'.
Nov 06 12:00:08 triton kernel: Bluetooth: hci0: read Intel version: 370710018002030d00
Nov 06 12:00:08 triton kernel: Bluetooth: hci0: Intel Bluetooth firmware file: intel/ibt-hw-37.7.10-fw-1.80.2.3.d.bseq
Nov 06 12:00:08 triton systemd-sleep[22492]: System resumed.
Nov 06 12:00:08 triton systemd[1]: Starting Load/Save RF Kill Switch Status...
Nov 06 12:00:08 triton systemd[1]: Started Suspend.
Nov 06 12:00:08 triton systemd[1]: sleep.target: Unit not needed anymore. Stopping.
Nov 06 12:00:08 triton systemd[1]: Reached target Bluetooth.
Nov 06 12:00:08 triton systemd[1]: Stopped target Sleep.
Nov 06 12:00:08 triton systemd[1]: Reached target Suspend.
Nov 06 12:00:08 triton systemd-logind[523]: Operation finished.
Nov 06 12:00:08 triton physlock[22491]: physlock: error: could not activate console # 8: Interrupted system call
```

Any ideas whats going wrong and how I can debug and fix this?

Thank you kindly

f

**Comment by mautz** [comment]
This is my working systemd file:

[Unit]
Description=Lock session using physlock

[Service]
User=[your_username]
ExecStart=/usr/bin/physlock
[Install]
WantedBy=sleep.target

**Comment by muennich** [comment]
Can you please try out the fix on top of branch `issue28`?

**Comment by muennich** [comment]
I am pretty sure that this issue has been fixed with commit d3a99b0. I am closing this issue to get on with the next stable release. Feel free to reopen, if you are still running into this issue.
